### PR TITLE
[SDK-3142] Add Vanilla JavaScript interactive quickstart

### PR DESCRIPTION
**Pull request by stevehobbsdev**
This PR adds the migrated vanilla JavaScript quickstart. Note that this is a complete rewrite, as the existing quickstart is of the older tutorial format as doesn't suit the new quickstart design.

⭐  Very happy to receive feedback on the code snippets, especially the "user profile" section and anything that would make it more concise while retaining context.

image

### CODE REVIEW

File: articles/quickstart/spa/vanillajs/interactive.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+- make sure the Auth0 client has been correctly configured with your Auth0 domain and client ID
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+
+:::
+::::
+
+## Handle the Callback from Auth0 {{{ data-action=code data-code="app.js#14:19" }}}
+
+To process the result from Auth0 when the browser is redirect back to your application, call the `handleRedirectCallback()` function on the Auth0 client.
```
Comment (adamjmcgrath) [inline articles/quickstart/spa/vanillajs/interactive.md]:
```suggestion
To process the result from Auth0 when the browser is redirected back to your application, call the `handleRedirectCallback()` function on the Auth0 client.
```

File: articles/quickstart/spa/vanillajs/interactive.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+---
+
+# Add Login to your JavaScript App
+
+Auth0 allows you to add authentication to almost any application type quickly. This guide demonstrates how to integrate Auth0, add authentication, and display user profile information in a Single Page Application that uses plain JavaScript, using the [Auth0 SPA SDK](https://github.com/auth0/auth0-spa-js).
+
+To use this quickstart, you’ll need to:
+
+- Sign up for a free Auth0 account or log in to Auth0.
+- Have a working React project that you want to integrate with. Alternatively, you can view or download a sample application after logging in.
```
Comment (adamjmcgrath) [inline articles/quickstart/spa/vanillajs/interactive.md]: ~~React~~

File: articles/quickstart/spa/vanillajs/interactive.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+## Add Login to Your Application {{{ data-action=code data-code="app.js#6:12" }}}
+
+The Auth0 SPA SDK gives you tools to quickly implement user authentication in your JavaScript application. Executing `loginWithRedirect()` redirects your users to the Auth0 Universal Login Page, where Auth0 can authenticate them. Upon successful authentication, Auth0 will redirect your users back to your application.
+
+Create a button in your application that calls `loginWithRedirect()` when clicked.
+
+::::checkpoint
+::: checkpoint-default
+Once a login button and the callback handler have been added, verify that you can click the button and are redirected to Auth0 for authentication, and that you do not receive any errors in the console relating to Auth0.
```
Comment (adamjmcgrath) [inline articles/quickstart/spa/vanillajs/interactive.md]: The callback handler has not been added at this point

File: articles/quickstart/spa/vanillajs/interactive.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+- check that the `redirect_uri` option has been configured to your application's URL
+- if you have an `error` query parameter, inspect it to learn the cause of the error
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+:::
+::::
+
+## Add Logout to Your Application {{{ data-action=code data-code="app.js#21:27" }}}
+
+Now that you can log in to your application, you need [a way to log out](/logout/guides/logout-auth0). The Auth0 client provides a `logout()` method that you can use to log a user out of your app. Executing `logout()` redirects your users to your [Auth0 logout endpoint](/api/authentication?javascript#logout) and then immediately redirects them back to your application.
```
Comment (adamjmcgrath) [inline articles/quickstart/spa/vanillajs/interactive.md:95]: Will the logout link go to `https://auth0.com/logout/guides/logout-auth0`? (404)

Comment (adamjmcgrath) [reply articles/quickstart/spa/vanillajs/interactive.md:95]: Same for `/api` - not sure if these automatically get the `/docs` prefix

Comment (stevehobbsdev) [reply articles/quickstart/spa/vanillajs/interactive.md:95]: Yes - this content was borrowed from [the Angular quickstart](https://auth0.com/docs/quickstart/spa/angular#add-logout-to-your-application) where you can verify that the links work.

**Review by adamjmcgrath — APPROVED** [review]

**Pull request merged**